src/anim/animation.js

```diff
@@ -6,6 +6,15 @@ Object.assign(pc, function () {
         this.scale = scale;
     };
 
+    var CKey = function CKey(time, value) {
+        this.time = time;
+        this.value = value;
+    };
+
+    CKey.POS = 0;
+    CKey.ROT = 1;
+    CKey.SCL = 2;

```

Comment:
These should be defined as constants. Either pc.KEYTYPE_POS, pc.KEYTYPE_ROT, pc.KEYTYPE_SCL or pc.KEYTYPE_POSITION, pc.KEYTYPE_ROTATION, pc.KEYTYPE_SCALE. It's a general coding style of the engine.